common/log/loggerimpl/logger_test.go

```diff
@@ -132,3 +133,25 @@ func TestEmptyMsg(t *testing.T) {
 	assert.Equal(t, out, `{"level":"info","msg":"`+defaultMsgForEmpty+`","error":"test error","wf-action":"add-workflow-started-event","logging-call-at":"logger_test.go:`+lineNum+`"}`+"\n")
 
 }
+
+func TestErrorWithDetails(t *testing.T) {
+	sb := &strings.Builder{}
+	zapLogger := zap.New(zapcore.NewCore(zapcore.NewJSONEncoder(zapcore.EncoderConfig{MessageKey: "msg"}), zapcore.AddSync(sb), zap.DebugLevel))
```

Comment:
There is a `zaptest` package that kinda covers this: https://pkg.go.dev/go.uber.org/zap@v1.21.0/zaptest

... but tbh it's sorta annoying to use, roughly as verbose as you have here or maybe a bit worse.  And in this case we do actually care about the json-encoded form, so I think this is fine 👍